### PR TITLE
DSD-615: Storybook arg names & cleanup

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-615](https://jira.nypl.org/browse/DSD-615)

## This PR does the following:
- Updates all component Storybook documentation by cleaning up unused variables and fixing typos.
- Add defaults to props in the Controls section.
- Cleans up stray console warning coming from a deprecated package.
- There is _one_ error coming from the emotion package which is not our fault - https://stackoverflow.com/questions/57239807/the-pseudo-class-first-child-is-potentially-unsafe-when-doing-server-side-ren
- It also updates how arg names for enums are displayed. It's confusing when current options are the values themselves in lowercase. For example, in the `StatusBadge` component, the `level` prop should have a value of `StatusBadgeTypes.Low`, `StatusBadgeTypes.Medium`, or `StatusBadgeTypes.High`:

```
<StatusBadge level={StatusBadgeTypes.Low} />
```

but Storybook, by default, shows the actual compiled value of the enum `StatusBadgeTypes.Low` as `"low"`. This causes confusing and causes devs to use this pattern instead:

```
<StatusBadge level="low" />
```

Technically, this is _still_ correct but we want to stick with the enum variable pattern.

Visual example before:
Screen Shot 2021-12-03 at 12 06 47 PM

after:
Screen Shot 2021-11-30 at 10 16 03 AM

**Once this is approved, I'll update the wiki page for "Anatomy of a Story".**

## How has this been tested?

Please go through every page/component in Storybook to make sure everything looks and works fine! The focus should be on the "args table" which displays the Controls and the "default value".

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] View [the example in Storybook]()

### CODE REVIEW

**Comment by EdwinGuzman** [comment]
One more thing:
It's also confusing looking at a component and not seeing _all_ available props. Instead of removing them from the Controls/ArgsTable section, they are just disabled and can't be controlled. I think this is better even though it makes the page longer.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr788-d7x3of4rzfc66kdugghfur2vqphitirq.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61aa4fc50048b8f1367b921e

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr788-d7x3of4rzfc66kdugghfur2vqphitirq.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61aa4fc50048b8f1367b921e

File: src/components/VideoPlayer/VideoPlayer.stories.mdx [hunk 1]
```diff
@@ -49,25 +59,39 @@ The `aspectRatio` prop is used to control the sizing of the space allotted for t
   <Story
     name="VideoPlayer"
     args={{
-      videoType: VideoPlayerTypes.Vimeo,
-      videoId: "",
-      headingText: "NYPL Video",
+      aspectRatio: "VideoPlayerAspectRatios.FourByThree",
```
Comment (bigfishdesign13) [inline src/components/VideoPlayer/VideoPlayer.stories.mdx]: The video that is first loaded has a 16:9 aspect ratio, so the `apectRatio` prop should match that. Change this to `SixteenByNine`.

**Review by bigfishdesign13 — COMMENTED** [review]
Some general comments.
- The props should be alphabetized on every story.
- Make sure `Default` column is populated with the default value set in component.
- Is there any way to increase the width of the `Default` column?  Eevn if there is a way to adjust that column width, I don't think there is enough width within the UI to give us what we would need to really make a difference.
- The description for any props that are for internal use should include a statement saying so.
- The values in the `Description` column are showing the end string value from the enums.  Can those values should the full enum value?  (ex. `ImageSizes.Medium)  That might be a bit much.

**Comment by EdwinGuzman** [comment]
> The props should be alphabetized on every story.

I tried to do this but because some props are declared on the global `argTypes` and some in the local `argTypes`, they get out of order. So you'll see that there might be sets of props alphabetized.

> Make sure `Default` column is populated with the default value set in component.

I did this. Not all props have a default value so many props are just `-`.

> Is there any way to increase the width of the Default column? 

I'll look into this.

> The description for any props that are for internal use should include a statement saying so.

Yes. Some props that are used only in Storybook say so but it's not clear that some props are only for internal use within the DS.

> The values in the Description column are showing the end string value from the enums. Can those values should the full enum value? (ex. `ImageSizes.Medium) That might be a bit much.

I couldn't find a way to do this. It is currently using the compiled enum value and we don't explicitly set it, so we may have to make custom configurations if we want this.

**Comment by EdwinGuzman** [comment]
> The values in the Description column are showing the end string value from the enums. Can those values should the full enum value? (ex. `ImageSizes.Medium) That might be a bit much.

This is sort of possible but now I get this:
Screen Shot 2021-12-08 at 11 07 45 AM

and this is how it was before:
Screen Shot 2021-12-08 at 11 10 10 AM


I'm leaning on the first one (even though each enum key value doesn't display). What do you think?

**Comment by bigfishdesign13** [comment]
> I'm leaning on the first one (even though each enum key value doesn't display). What do you think?

I also like the first option, but I think the description should now say something about the prop value being an enum.  What is the technical way to talk about enum values?

**Comment by EdwinGuzman** [comment]
@bigfishdesign13 update:

> The props should be alphabetized on every story.

Done, but now every prop must be listed in the `args` prop and in alphabetical order. Also, if they don't have a value, `undefined` should be used. Check out the stories for examples.

> Is there any way to increase the width of the Default column? Eevn if there is a way to adjust that column width, I don't think there is enough width within the UI to give us what we would need to really make a difference.

The best I could manually override was only the third column. It seems like the other columns are dynamic and CSS overrides don't affect them...

> The description for any props that are for internal use should include a statement saying so.

Done. They're mostly in caps lock (I might've missed a few so please point them out).
Screen Shot 2021-12-10 at 10 36 33 AM

> The values in the Description column are showing the end string value from the enums. Can those values should the full enum value? (ex. `ImageSizes.Medium) That might be a bit much.

I think a different ticket to explain the enum/object usage in a separate "how-to" or typescript docs would be helpful.

**Comment by EdwinGuzman** [comment]
Another update:

Check out the stories for `Form`, `Image`, and `HorizontalRule`. Because we are using the native HTML element props and creating a union with our own props, _all_ the props are listed in storybook! For example, in the `HorizontalRule` we are declaring props as `React.ComponentProps<"hr"> & HorizontalRuleProps` and this makes it list over 259 props, only four of which we declared for the DS component. I don't think having the `React.ComponentProps<"...">` is beneficial for us and if a dev needs to pass in a specific HTML attribute, they can let us know why. Or, we can return to thinking about using the `attributes` prop again. I think for now we can remove it until necessary.

I did try to find a way to remove those props from Storybook but couldn't find a way. It may be with the latest update from them.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr788-d7x3of4rzfc66kdugghfur2vqphitirq.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61aa4fc50048b8f1367b921e

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr788-d7x3of4rzfc66kdugghfur2vqphitirq.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61aa4fc50048b8f1367b921e

**Comment by EdwinGuzman** [comment]
Update: will create a ticket to fix the props for `Form`, `Image`, and `HorizontalRule`.

**Comment by bigfishdesign13** [comment]
@hokei and @nypl-wluisi 
Can you both take a look at this PR?  Thanks.

**Comment by EdwinGuzman** [comment]
Nudge

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr788-d7x3of4rzfc66kdugghfur2vqphitirq.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61aa4fc50048b8f1367b921e

**Comment by EdwinGuzman** [comment]
@bigfishdesign13  updated from `development`, can you re-check the stories for `Card`, `Image`, and `VideoPlayer`? They had recent updates which I want to make sure are correct with this PR's updates.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr788-d7x3of4rzfc66kdugghfur2vqphitirq.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61aa4fc50048b8f1367b921e

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr788-d7x3of4rzfc66kdugghfur2vqphitirq.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61aa4fc50048b8f1367b921e

**Comment by EdwinGuzman** [comment]
Just a reminder for myself to update the `Toggle` component's story.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr788-d7x3of4rzfc66kdugghfur2vqphitirq.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61aa4fc50048b8f1367b921e

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr788-d7x3of4rzfc66kdugghfur2vqphitirq.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61aa4fc50048b8f1367b921e

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr788-d7x3of4rzfc66kdugghfur2vqphitirq.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61aa4fc50048b8f1367b921e

File: src/components/VideoPlayer/VideoPlayer.stories.mdx [hunk 2]
```diff
@@ -51,15 +74,39 @@ The `aspectRatio` prop is used to control the sizing of the space allotted for t
...
         "Fusce dapibus, tellus ac cursus commodo, tortor mauris condimentum nibh, ut fermentum massa justo sit amet risus. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus.",
       embedCode: `<iframe src="https://player.vimeo.com/video/421404144?h=5467db7edd" width="640" height="360" frameborder="0" allow="autoplay; fullscreen; picture-in-picture" allowfullscreen></iframe>`,
       headingText: "NYPL Video",
       helperText:
         "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vivamus sagittis lacus vel augue laoreet rutrum faucibus dolor auctor.",
+      id: "videoPlayer-id",
+      iframeTitle: undefined,
+      showHelperInvalidText: true,
+      useEmbed: true,
+      videoId: "474719268",
```
Comment (bigfishdesign13) [inline src/components/VideoPlayer/VideoPlayer.stories.mdx]: `videoId` and `videoType` should no longer be set for the initial story.  Only `embedCode` should have a value.

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
The story for `Video` needs some adjustments.  Specifically, some of the props need to be altered.  I am also seeing that the `aspectRatio` prop is not functional, but at this point I am not sure why.  It is working for `VideoPlayer` in the `development` branch.

**Comment by EdwinGuzman** [comment]
Updated.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr788-d7x3of4rzfc66kdugghfur2vqphitirq.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61aa4fc50048b8f1367b921e

**Review by nypl-wluisi — COMMENTED** [review]
@EdwinGuzman @bigfishdesign13 This looks pretty good. There's a handful of stories that have this msg: _"This story is not configured to handle controls. Learn how to add controls"._ Some examples:

https://pr788-d7x3of4rzfc66kdugghfur2vqphitirq.tugboat.qa/?path=/story/components-form-elements-select--labelling-variations

https://pr788-d7x3of4rzfc66kdugghfur2vqphitirq.tugboat.qa/?path=/story/components-form-elements-button--button-groups

https://pr788-d7x3of4rzfc66kdugghfur2vqphitirq.tugboat.qa/?path=/story/components-basic-elements-hero--color-variations-for-secondary-hero

https://pr788-d7x3of4rzfc66kdugghfur2vqphitirq.tugboat.qa/?path=/story/components-basic-elements-card--card-image-position

I dont know if these are part of this scope of work?

**Comment by EdwinGuzman** [comment]
> There's a handful of stories that have this msg: "This story is not configured to handle controls. Learn how to add controls". 

@nypl-wluisi that's correct. In this PR, and moving forward, only the first story in each section will have Controls enabled. This is indicated by the name. For example, in the first link shared, it is for the `Select` and points to the "Labelling Variations". The first story above it, "Select with Controls", is the main story that has it enabled. This is the _same_ pattern for all components. We're hoping this makes it easier to (1) manipulate a component with _all_ its props in only one story, and (2) to write concise mdx storybook files.

File: src/components/VideoPlayer/VideoPlayer.stories.mdx [hunk 3]
```diff
@@ -20,15 +26,22 @@ import { getCategory } from "../../utils/componentCategories";
...
+    aspectRatio: {
+      control: { type: "select" },
+      options: ratiosEnumValues.options,
+    },
+    className: { control: false },
+    id: { control: false },
+    showHelperInvalidText: {
+      table: { defaultValue: { summary: true } },
+    },
+    videoId: { control: false },
```
Comment (bigfishdesign13) [inline src/components/VideoPlayer/VideoPlayer.stories.mdx]: `videoId` and `videoType` don't need to be disabled.  They just need to be null initially.  Someone viewing the `Video` story should be able to update the props.

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Sorry, I may not have stated my comment clearly yesterday.  `videoId` and `videoType` should still be enabled.  They just need to be null initially.

**Review by nypl-wluisi — APPROVED** [review]
looks good to me

File: src/components/VideoPlayer/VideoPlayer.stories.mdx (hunk 3 above)
Comment (EdwinGuzman) [reply src/components/VideoPlayer/VideoPlayer.stories.mdx]: Updated.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr788-d7x3of4rzfc66kdugghfur2vqphitirq.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61aa4fc50048b8f1367b921e

**Review by bigfishdesign13 — APPROVED** [review]
Very nicely done.

**Comment by EdwinGuzman** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr788-d7x3of4rzfc66kdugghfur2vqphitirq.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/61aa4fc50048b8f1367b921e

**Pull request merged**